scripts/commands/regenerateApiDocs.ts

```diff
@@ -0,0 +1,158 @@
+      default: false,
+      description: "Regenerate prior releases?",
+    })
+    .parseSync();
+};
+
+zxMain(async () => {
+  const args = readArgs();
+
+  const initialStatus = await gitStatus();
+  if (initialStatus !== "") {
+    console.error(`
+        Repository must have clean Git state when calling
+        \`npm run regenerate-apis\`. 'git status' returned:\n\n${initialStatus}`);
+    process.exit(1);
+  }
+
+  const currentBranch = await gitBranch();
+  if (currentBranch == "main\n") {
+    console.error(`
+        Please create a dedicated branch to regenerate the API docs correctly.
+        Use 'git checkout -b <name-branch>' to continue`);
+    process.exit(1);
+  }
+
+  const results = new Map<string, string[]>();
+  for (const pkgName of Pkg.VALID_NAMES) {
+    if (args.package && pkgName != args.package) {
+      continue;
+    }
+
+    const pkgVersions = await getPackageVersions(pkgName, args.historical);
+    const result = await regenerateVersions(pkgName, pkgVersions);
+    results.set(pkgName, result);
+  }
+
+  console.log("");
+  results.forEach((result: string[], pkgName: string) => {
+    console.log(`Regeneration of ${pkgName}:`);
+    result.forEach((msg) => console.error(msg));
+    console.log("");
+  });
+});
+
+async function regenerateVersions(
+  pkgName: string,
+  versions: string[],
+): Promise<string[]> {
+  const results: string[] = [];
+
+  for (const [index, version] of versions.entries()) {
+    try {
+      if (index == versions.length - 1) {
+        // The last version is always the latest stable minor release
+        await $`npm run gen-api -- -p ${pkgName} -v ${version}`;
+      } else {
+        await $`npm run gen-api -- -p ${pkgName} -v ${version} --historical`;
+      }
+
+      if ((await gitStatus()) !== "") {
+        await gitCommit(`Regenerate ${pkgName} ${version}`);
+        results.push(`✅ ${pkgName} ${version} regenerated correctly`);
+      } else {
+        results.push(`☑️ ${pkgName} ${version} is up-to-date`);
+      }
+    } catch (_) {
+      await gitRestore(".");
+      results.push(`❌ ${pkgName} ${version} failed to regenerate`);
+    }
+  }
+
+  return results;
+}
+
+async function getPackageVersions(
+  pkgName: string,
+  historical: boolean,
+): Promise<string[]> {
+  const pkgDocsPath = `docs/api/${pkgName}`;
+  const historicalVersions: string[] = [];
+
+  if (historical) {
+    const historicalFolders = (
+      await readdir(`${pkgDocsPath}`, { withFileTypes: true })
+    ).filter((file) => file.isDirectory() && file.name.match(/[0-9].*/));
+
+    for (const folder of historicalFolders) {
+      const historicalVersion = JSON.parse(
+        fs.readFileSync(`${pkgDocsPath}/${folder.name}/_package.json`, "utf-8"),
+      );
+      historicalVersions.push(historicalVersion.version);
+    }
+  }
+
+  const currentVersion = JSON.parse(
+    fs.readFileSync(`${pkgDocsPath}/_package.json`, "utf-8"),
+  );
+  return [...historicalVersions, currentVersion.version];
```

Comment:
If you take my above suggestion, then you could either return a tuple or an object if you want it more structured.